pyface/data_view/value_types/enum_value.py

```diff
@@ -0,0 +1,162 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import Callable, List
+
+from .editable_value import EditableValue
+
+
+class EnumValue(EditableValue):
+    """ Editable value that takes one of a collection of pre-set values.
+
+    Each value can be associated with text, colors and images by supplying
+    functions ``format``, ``colors`` and ``images``, respectively.
+    """
+
+    #: The list of values which are allowed for the value.
+    values = List()
```

Comment:
No problems with dynamic values unless the user wants them to update while the editor is open and actively being edited, which I don't plan to support (I can't think of a valid use-case).

There is an issue of what if this changes and makes the current value invalid (or if the data model changes and makes the current value invalid), but we potentially have that issue in a bunch of places with existing value types.  My feeling is that we let that happen rather than trying to force a valid value.